### PR TITLE
View Renderer interface that could render any view

### PR DESCRIPTION
**Pull request by sunayna**

### CODE REVIEW

File: src/java/com/sarality/app/view/list/ViewRenderer.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.sarality.app.view.list;
```
Comment (abhideep) [inline src/java/com/sarality/app/view/list/ViewRenderer.java]: Move this out of list and into view

File: src/java/com/sarality/app/view/list/BaseViewRenderer.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Base View Renderer to render any kind of view
+ *
+ * @author sunayna@ (Sunayna Uberoy)
+ */
+public abstract class BaseViewRenderer<T> implements ViewRenderer<T> {
+  protected final Context context;
+  protected final View view;
+
+  public BaseViewRenderer(Context context, View view) {
```
Comment (abhideep) [inline src/java/com/sarality/app/view/list/BaseViewRenderer.java]: if the render method already takes in a View why do we need to take here in the Constructor?

Comment (sunayna) [inline src/java/com/sarality/app/view/list/BaseViewRenderer.java]: i am not going to check this one in yet. Let me see if we need this at all.

Comment (abhideep) [inline src/java/com/sarality/app/view/list/BaseViewRenderer.java]: Makes sense. If and when you do add it, we may also want to add the ability to registerActions and setupActions in the ViewRenderer.

**Pull request merged**